File: ios/Plugin/DetectEnvironmentPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(DetectEnvironmentPlugin, "DetectEnvironment",
            CAP_PLUGIN_METHOD(isDebug, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(isTestFlight, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(isAdHoc, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(isSimulator, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(isAppStore, CAPPluginReturnPromise);
            CAP_PLUGIN_METHOD(isSandbox, CAPPluginReturnPromise);
)

File: ios/Plugin/DetectEnvironmentPlugin.swift
import Capacitor
import Foundation

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(DetectEnvironmentPlugin)
public class DetectEnvironmentPlugin: CAPPlugin {
    private let implementation = DetectEnvironment()

    @objc func isDebug(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.isDebug(),
        ])
    }

    @objc func isTestFlight(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.isTestFlight(),
        ])
    }

    @objc func isAdHoc(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.isAdHoc(),
        ])
    }

    @objc func isSimulator(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.isSimulator(),
        ])
    }

    @objc func isAppStore(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.isAppStore(),
        ])
    }

    @objc func isSandbox(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.isSandbox(),
        ])
    }
}
